src/main/scala/edu/ie3/simona/agent/grid/DBFSAlgorithm.scala

```diff
@@ -404,7 +407,16 @@ trait DBFSAlgorithm extends PowerFlowSupport with GridResultsSupport {
                   sender ! GridPowerResponse(exchangePowers)
                   simulateGrid(updatedGridAgentBaseData, currentTick)
 
-                case _: FailedNewtonRaphsonPFResult =>
+                case failedResult: FailedNewtonRaphsonPFResult =>
+                  if (failedResult.cause == MaxIterationsReached) {
+                    ctx.log.warn(
+                      "Power flow in grid {} failed after {} iterations. Cause: {}",
+                      ctx.self,
+                      failedResult.iteration,
+                      failedResult.cause,
+                    )
+                  }
+
```

Comment:
I am not really sure if this is what we want. If I recall correctly, `FailedPowerFlow` is supposed to "bubble up" towards the slack GridAgent, who then decides to quit DBFS simulation. But it looks like the information is not forwarded in `goToPowerFlowCalculationOrStay`.

So if everything actually worked, I think only the slack GridAgent should print out a warning that some subgrid has failed powerflow calculation. Additionally each GA in which Newton-Raphson actually failed, could print out a warning as well.